### DISCUSSION TITLE
Does Edgetx use freeRTOS only for simulator or radio also?

### DISCUSSION THREAD

**Discussion by sanjuchopracool** [EdgeTX/edgetx#1833] [Q&A] [answered]
I was checking edgetx source code, specially rtos part. 
in rtos.h file, RTOS_CREATE_TASK internally call pthread API. I don't see any API related to freeRTOS task creation (xTaskCreate)

```
  inline void RTOS_CREATE_TASK(pthread_t &taskId, void * (*task)(void *), const char * name)
  {
    pthread_create(&taskId, nullptr, task, nullptr);
#ifdef __linux__
    pthread_setname_np(taskId, name);
#endif
  }
```

does edgetx use FreeRTOS only for simulator or for actual hardware also ?
If freeRTOS is used for actual hardware also, Please point out reference to files and code to check that code.

**Comment by raphaelcoeffic** [comment, marked as answer]
It is the opposite: it is used for hardware but not for simu. I‘ve been working on using it for simu as well, but there are some issues on macOS and probably on Windows as well. So I will have to find another solution.

The freeRTOS functions used on hardware can be found in the second part of the same file:
https://github.com/EdgeTX/edgetx/blob/ddf011ff12113d6d9617ee11019ac7d5873abce0/radio/src/rtos.h#L140-L267
